### PR TITLE
Use component label everywhere

### PR DESCRIPTION
**Pull request by lkysow**
Also add a test to ensure every file has it.

Fixes https://github.com/hashicorp/consul-k8s/issues/673

Checklist:
- [x] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/connect-inject-leader-election-role.yaml [hunk 1]
```diff
@@ -9,7 +9,7 @@ metadata:
     chart: {{ template "consul.chart" . }}
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
-    component: controller
+    component: connect-injector
```
Comment (ishustava) [inline charts/consul/templates/connect-inject-leader-election-role.yaml:12]: great catch!

File: charts/consul/templates/gossip-encryption-autogenerate-job.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ metadata:
     chart: {{ template "consul.chart" . }}
     heritage: {{ .Release.Service }}
     release: {{ .Release.Name }}
+    component: gossip-encryption-autogeneneration
```
Comment (ishustava) [inline charts/consul/templates/gossip-encryption-autogenerate-job.yaml:16]: To match the job name:
```suggestion
    component: gossip-encryption-autogenenerate
```

File: charts/consul/test/unit/helpers.bats [hunk 3]
```diff
@@ -119,6 +119,20 @@ load _helpers
...
+#
+# This test ensures that we set a "component: <blah>" in every file.
+#
+# If this test fails, you're likely missing setting that label somewhere.
+
+@test "helper/component-label: used everywhere" {
+  cd `chart_dir`
+  # Grep for files that don't have 'component: ' in them
+  local actual=$(grep -L 'component: ' templates/*.yaml | tee /dev/stderr )
+  [ "${actual}" = '' ]
```
Comment (ishustava) [inline charts/consul/test/unit/helpers.bats:133]: nice! that good ol' grep test 😄

**Review by ishustava — APPROVED** [review]
Thank you for fixing this!

**Review by david-yu — APPROVED** [review]
LGTM

**Pull request merged**